### DISCUSSION TITLE
Preventing `max_active_runs_per_dag` limit hit during long backfill operations

### DISCUSSION THREAD

**Discussion by huymq1710** [apache/airflow#34390] [Q&A]
I am currently using Airflow with the **KubernetesExecutor** and have set the `max_active_runs_per_dag: 8`. In my workflow, I am running `airflow dags backfill` operations using the LocalExecutor which can take a considerable amount of time.

I've noticed that during these long backfill operations, I am hitting the `max_active_runs_per_dag` limit. This is preventing new DAGs for new dates from executing as expected.

I am looking for suggestions on how to prevent this scenario. Is there a way to manage backfill operations such that they don't contribute towards the max_active_runs_per_dag limit? Or perhaps there are best practices for managing backfill operations to avoid this issue?

Any advice or insights from the community would be greatly appreciated!

Thank you,

**Comment by namndh** [comment]
Meeting the same issue. Looking forward for optimal approaches.

**Comment by Taragolis** [comment]
Airflow is primarily a community-driven project, and most of its community members are also active users of the platform. At that moment we looking for volunteers who might want to help [refactor backfill_job_runner](https://github.com/apache/airflow/issues/33249). 
- [Contributor's Quick Start](https://github.com/apache/airflow/blob/main/CONTRIBUTORS_QUICK_START.rst#contributors-quick-start)
- [CONTRIBUTING.rst](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst)


<details>
  <summary>Some pain under the spoiler 🥲😢😭😢🥲</summary>
  
![image](https://github.com/apache/airflow/assets/3998685/ff9bd43c-cd80-4687-94f5-6e545ad07a4e)
  
</details>

**Comment by huymq1710** [comment]
cc: @potiuk 
How about provide a flag `--max-active-dag-runs` for backfill ?

**Reply by boushphong** [reply]
Yes, please. But I think the refactoring part has a higher priority right now?
